EditorInterface.cpp

```diff
@@ -2986,6 +3095,154 @@ void EditorInterface::IGSquadEditor()
+											   get_bit(ckey->flags, 5), get_bit(ckey->flags, 6), get_bit(ckey->flags, 7), get_bit(ckey->flags, 8) };
+
+					
+					if (ImGui::Button("Flags")) {
+						ImGui::OpenPopup("ChoreoKey flags");
+						ckflags[0] = get_bit(ckey->flags, 1);
+						ckflags[1] = get_bit(ckey->flags, 2);
+						ckflags[2] = get_bit(ckey->flags, 3);
+						ckflags[3] = get_bit(ckey->flags, 4);
+						ckflags[4] = get_bit(ckey->flags, 5);
+						ckflags[5] = get_bit(ckey->flags, 6);
+						ckflags[6] = get_bit(ckey->flags, 7);
+						ckflags[7] = get_bit(ckey->flags, 8);
+					}
+					ImGui::SameLine();
+					static int ChoreoDeleteNum = -1;
+					static int ChoreoKDeleteNum = -1;
+					static bool AdjustChoreo = false;
+					static bool AdjustPrevChoreoSlots = false;
+					ImGui::PushStyleColor(ImGuiCol_Button, (ImVec4)ImColor::HSV(0.0f, 1.0f, 0.8f));
+					ImGui::PushStyleColor(ImGuiCol_ButtonHovered, (ImVec4)ImColor::HSV(0.0f, 1.0f, 0.9f));
+					ImGui::PushStyleColor(ImGuiCol_ButtonActive, (ImVec4)ImColor::HSV(0.0f, 1.0f, 1.0f));
+					if (ImGui::Button("Delete Tools")) {
+						ImGui::OpenPopup("Delete Options");
+					}
+					ImGui::PopStyleColor(3);
+					if (ImGui::BeginPopup("ChoreoKey flags")) {
+
+						ImGui::Checkbox("Don't rotate around asterix", &ckflags[0]);
+						ImGui::Checkbox("Bit 2", &ckflags[1]);
+						ImGui::Checkbox("Bit 3", &ckflags[2]);
+						ImGui::Checkbox("Formations always have spears out", &ckflags[3]);
+						ImGui::Checkbox("Look at asterix", &ckflags[4]);
+						ImGui::Checkbox("Bit 6", &ckflags[5]);
+						ImGui::Checkbox("Enemies can run", &ckflags[6]);
+						ImGui::Checkbox("Bit 8", &ckflags[7]);
+
+						if (ImGui::Button("OK")) {
+							for (int8_t index = 0; index < 8; index++) {
+								ckey->flags = toggle_bit(ckey->flags, index, ckflags[index]);
+							}
+						}
+						ImGui::EndPopup();
+					}
+
+					if (ImGui::BeginPopup("Delete Options")) {
+
+						ImGui::InputInt("Delete num ChoreoGraphy:", &ChoreoDeleteNum);
+						ImGui::Checkbox("Merge Choreokeys to previous ChoreoGraphy", &AdjustPrevChoreoSlots);
+						//ImGui::Checkbox("Adjust Number of Choreokeys", &AdjustChoreo);
+						ImGui::InputInt("Delete num ChoreoKey:", &ChoreoKDeleteNum);
+						ImGui::Checkbox("Adjust Number of Choreokeys", &AdjustChoreo);
+
+						if (ImGui::Button("OK")) {
+							if (selectedSquad) {
+								IsDoingHomeWork = true; // Extremely Important! PogChamp
+
+								if (ChoreoKDeleteNum > -1) {
+									//ChoreoKey
+									auto* Keytoremove = selectedSquad->choreoKeys[ChoreoKDeleteNum].get();
+									auto ChoreoKDeletePos = selectedSquad->choreoKeys.begin() + ChoreoKDeleteNum;
+									selectedSquad->choreoKeys.erase(ChoreoKDeletePos);
+									kenv.removeObject(Keytoremove);
+									if (AdjustChoreo == true) {
+										selectedSquad->choreographies[getChoreo(ChoreoKDeleteNum)]->numKeys = selectedSquad->choreographies[getChoreo(ChoreoKDeleteNum)]->numKeys - 1;
+									}
+								}
```

Comment:
Cool there is an option for adjusting choreo keys, but what happens if it is disabled? Last choreo will have one key more that no longer exists? That might cause a crash in future.

Reply:
Yeah, i left this in for the future if i ever need to manually insert a ChoreoKey with the python editor, but now i don't need to do that anymore 😉 it will automatically do that now, no enabling the box needed! Fixed 👍